static/app/views/eventsV2/table/arithmeticInput.tsx

```diff
@@ -134,17 +127,52 @@ export default class ArithmeticInput extends PureComponent<Props, State> {
     this.setState({dropdownVisible: false});
   };
 
+  makeActive(selection: number) {
+    const {options} = this.props;
+    const {partialTerm} = this.state;
+    const newOptionGroups = makeOptions(options, partialTerm);
+
+    for (const group of newOptionGroups) {
```

Comment:
I think this is a touch confusing, there are a low number of options here (low n) and the options are references so I'd recommend maybe doing something like the following to improve the clarity.

```js
const allOptions = newOptionGroups.reduce((prev, curr) => prev.push(...curr.options), []);
allOptions[selection].active = true;
```
You could also move it to a class getter to share with the function below, or alternatively memoize it inside a function to improve performance if it's a concern.

Reply:
That's not quite the same thing because your suggestion is flattening the groups into a single list whereas the intent is to keep them grouped. 

Definitely agree that this is a little confusing. I'll take a look 👍

Reply:
@Zylphrex no you can still return the `newOptionGroups` though, it's just to select the option to make active more simply.

Reply:
Oh I see what you mean 👍